src/main/java/lotto/domain/Rank.java

```diff
@@ -0,0 +1,54 @@
+package lotto.domain;
+
+import lotto.startegy.MatchStrategy;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+public enum Rank {
+    NONE(2, 0, count -> (count <= 2)),
+    FOURTH(3, 5000, count -> (count == 3)),
+    THIRD(4, 50_000, count -> (count == 4)),
+    SECOND(5, 1_500_000, count -> (count == 5)),
+    FIRST(6, 2_000_000_000, count -> (count == 6));
```

Comment:
괄호가 없어도 좋을 것 같아요 :) 취향이니 참고만 해주세요 😄 

```suggestion
    NONE(2, 0, count -> count <= 2),
    FOURTH(3, 5000, count -> count == 3),
    THIRD(4, 50_000, count -> count == 4),
    SECOND(5, 1_500_000, count -> count == 5),
    FIRST(6, 2_000_000_000, count -> count == 6);
```